skimage/util/shape.py

```diff
@@ -215,13 +217,18 @@ def view_as_windows(arr_in, window_shape, step=1):
     # -- basic checks on arguments
     if not isinstance(arr_in, np.ndarray):
         raise TypeError("`arr_in` must be a numpy ndarray")
-    if not isinstance(window_shape, tuple):
-        raise TypeError("`window_shape` must be a tuple")
+
+    if isinstance(window_shape, numbers.Number):
+        window_shape = tuple([window_shape] * arr_in.ndim)
     if not (len(window_shape) == arr_in.ndim):
         raise ValueError("`window_shape` is incompatible with `arr_in.shape`")
 
-    if step < 1:
-        raise ValueError("`step` must be >= 1")
+    if isinstance(step, numbers.Number):
+        if step < 1:
+            raise ValueError("`step` must be >= 1")
+        step = tuple([step] * arr_in.ndim)
```

Comment:
`(step,) * arr_in.ndim`

In fact, because `arr_in.ndim` is used so often, it might be better to make one call at the start, `ndim = arr_in.ndim`, and then use `ndim` for the rest of the function.